Source/WebCore/page/NavigateEvent.cpp

```diff
@@ -52,12 +53,63 @@ Ref<NavigateEvent> NavigateEvent::create(const AtomString& type, const NavigateE
     return adoptRef(*new NavigateEvent(type, init));
 }
 
-void NavigateEvent::intercept(NavigationInterceptOptions&&)
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigateevent-perform-shared-checks
+ExceptionOr<void> NavigateEvent::sharedChecks()
 {
+    if (!isTrusted())
+        return Exception { ExceptionCode::SecurityError, "Event is not trusted"_s };
+
+    if (defaultPrevented())
+        return Exception { ExceptionCode::InvalidStateError, "Event was already canceled"_s };
+
+    return { };
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigateevent-intercept
+ExceptionOr<void> NavigateEvent::intercept(NavigationInterceptOptions&& options)
+{
+    auto checkResult = sharedChecks();
+    if (checkResult.hasException())
```

Comment:
In modern C++ these 2 lines can be combined to one if IIRC.

Reply:
There are a few ways to format this but I'm sure what you intended. `checkResult` isn't truthy if an exception is set so `if (auto checkResult = sharedChecks())` doesn't work unfortunately.

Reply:
That is one of the things I like about modern C++ :) For example:
https://skebanga.github.io/if-with-initializer/